lib/index.js

```diff
@@ -118,15 +101,6 @@ function encodeArrayBuffer(packet, supportsBinary, callback) {
  */
 
 exports.encodeBase64Packet = function(packet, callback){
-
-  if (!Buffer.isBuffer(packet.data)) {
-    var buf = new Buffer(packet.data.byteLength);
-    for (var i = 0; i < buf.length; i++) {
-      buf[i] = packet.data[i];
-    }
-    packet.data = buf;
-  }
```

Comment:
basically yes, but it's exported. That's what I worry.

And I noticed the browser version of `encodeBase64Packet` accepts ArrayBuffer.
https://github.com/socketio/engine.io-parser/blob/master/lib/browser.js#L186

Maybe we shouldn't change this for now?